packages/jaeger-ui/src/model/transform-trace-data.tsx

```diff
@@ -135,6 +135,15 @@ export default function transformTraceData(data: TraceData & { spans: SpanData[]
     span.relativeStartTime = span.startTime - traceStartTime;
     span.depth = depth - 1;
     span.hasChildren = node.children.length > 0;
+    // Get the childSpanIds sorted based on endTime without changing tree structure
```

Comment:
L121 already builds a tree and sorts the children. I suggest we store that tree in the Trace object, instead of adding more arrays. The tree is also easier to use because it represents children as Nodes with direct access to Span, instead of requiring an extra lookup by ID

Reply:
Here in L121 line it builds a tree and sorts the children based on startTime.
And in that tree each node is of type `TreeNode`
https://github.com/jaegertracing/jaeger-ui/blob/7a51ede7803c432825b29066e3a31e80cb2f8ef2/packages/jaeger-ui/src/utils/TreeNode.js#L28-L31
And in a TreeNode there are only two parameters `value` and `children` where value means a string (current spanId) and children means an array of TreeNode (children of current span) 
so even if we store tree in the trace object also we should again lookup by ID as we don't have direct access to span